### ISSUE TITLE
Extend parser functionality

### ISSUE THREAD

**Issue by bumbeishvili** [rhysd/node-github-trend#2]
Right now parser only returns trending repository name and owner info in following format
```javascript
{
     name:"Repo name",
     owner:"Repo owner"
}
```
although it's possible to extract more info (without additional api calls)

I have implemented method to extract following data format
```javascript
{
      name: null,
      owner: null,
      description: null,
      language: null,
      allStars: null,
      todaysStars: null,
}
```

add method body
```javascript
    Scraper.prototype.scrapeTrendingReposFullInfo = function (lang_name) {
        return this.fetchTrendPage(lang_name).then(function (html) {
            var dom = cheerio.load(html);
            return dom(".repo-list li")
                .toArray()
                .map(function (li, i) {
                    var result = {
                        index: i,
                        name: null,
                        owner: null,
                        description: null,
                        language: null,
                        allStars: null,
                        todaysStars: null,
                    }
                    //extract owner and repo name
                    var domElem = dom(li);
                    var a = domElem.find('h3 a').toArray()[0];

                    var href = a.attribs["href"];
                    var match = href.match(RE_HREF_SCRAPE);

                    if (!match) {
                        console.log("Invalid repo: " + href);
                    }

                    result.owner = match[1];
                    result.name = match[2];

                    //extract description
                    var p = domElem.find('p').toArray()[0];
                    if (p) {
                        result.description = p.children[0].data
                    }

                    //extract programming language
                    var lang = domElem.find('[itemprop="programmingLanguage"]').toArray()[0];
                    if (lang) {
                        result.language = lang.children[0].data
                    }

                    //extract all stars
                    var allStars = domElem.find('[aria-label="Stargazers"]').toArray()[0];
                    if (allStars) {
                        result.allStars = allStars.children[2].data
                    }

                    //extract todays stars
                    var todaysStars = domElem.find('.text-gray span:last-child').toArray()[0];
                    if (todaysStars) {
                        result.todaysStars = todaysStars.children[2].data
                    }


                    //clean result
                    var emptyStringmatcher = /^\s+|\s+$/g;

                    var keys = Object.keys(result);

                    keys.forEach(k => {
                        if (typeof result[k] === 'string' || result[k] instanceof String) {
                            result[k] = result[k].replace(emptyStringmatcher, '');
                        }
                    });

                    return result;
                });
        });
    };
```

Invoking is same
```javascript
scraper.scrapeTrendingReposFullInfo('csharp')
                .then(repos => {
                    console.log(repos)
                 });
```

and result is following
```json
[
   {
      "index":0,
      "name":"shadowsocks-windows",
      "owner":"shadowsocks",
      "description":"If you want to keep a secret, you must also hide it from yourself.",
      "language":"C#",
      "allStars":"7,586",
      "todaysStars":"23 stars today"
   },
   {
      "index":1,
      "name":"BaiduPanDownload",
      "owner":"Mrs4s",
      "description":"百度网盘不限速下载工具",
      "language":"C#",
      "allStars":"2,296",
      "todaysStars":"19 stars today"
   },
   {
      "index":2,
      "name":"dapper-dot-net",
      "owner":"StackExchange",
      "description":"Dapper - a simple object mapper for .Net",
      "language":"C#",
      "allStars":"5,602",
      "todaysStars":"14 stars today"
   },
   {
      "index":3,
      "name":"corefx",
      "owner":"dotnet",
      "description":"This repo contains the .NET Core foundational libraries, called CoreFX. It includes classes for collections, file systems, console, XML, async and many others.",
      "language":"C#",
      "allStars":"11,135",
      "todaysStars":"11 stars today"
   },
   {
      "index":4,
      "name":"Lite.Abstractions",
      "owner":"AspectCore",
      "description":"AspectCore Lite is an lightweight aop lib for Asp.Net Core Framework.",
      "language":"C#",
      "allStars":"23",
      "todaysStars":"13 stars today"
   },
   {
      "index":5,
      "name":"LightBulb",
      "owner":"Tyrrrz",
      "description":"Background application that adjusts screen gamma as the day goes, reducing blue light when it gets late (C#/WPF/MVVM)",
      "language":"C#",
      "allStars":"83",
      "todaysStars":"11 stars today"
   },
   {
      "index":6,
      "name":"shadowsocksr-csharp",
      "owner":"shadowsocksr",
      "description":"ShadowsocksR for Windows",
      "language":"C#",
      "allStars":"1,358",
      "todaysStars":"11 stars today"
   },
   {
      "index":7,
      "name":"BotBuilder",
      "owner":"Microsoft",
      "description":"The Microsoft Bot Builder SDK is one of three main components of the Microsoft Bot Framework. The Microsoft Bot Framework provides just what you need to build and connect intelligent bots that interact naturally wherever your users are talking, from text/SMS to Skype, Slack, Office 365 mail and other popular services.",
      "language":"C#",
      "allStars":"3,387",
      "todaysStars":"10 stars today"
   },
   {
      "index":8,
      "name":"Radarr",
      "owner":"Radarr",
      "description":"A fork of Sonarr to work with movies à la Couchpotato.",
      "language":"C#",
      "allStars":"231",
      "todaysStars":"11 stars today"
   },
   {
      "index":9,
      "name":"hakchi2",
      "owner":"ClusterM",
      "description":"NES Mini pimp tool",
      "language":"C#",
      "allStars":"123",
      "todaysStars":"10 stars today"
   },
   {
      "index":10,
      "name":"CefSharp",
      "owner":"cefsharp",
      "description":".NET (WPF and Windows Forms) bindings for the Chromium Embedded Framework",
      "language":"C#",
      "allStars":"2,682",
      "todaysStars":"7 stars today"
   },
   {
      "index":11,
      "name":"PowerShell",
      "owner":"PowerShell",
      "description":"PowerShell for every system!",
      "language":"C#",
      "allStars":"5,971",
      "todaysStars":"8 stars today"
   },
   {
      "index":12,
      "name":"MaterialDesignInXamlToolkit",
      "owner":"ButchersBoy",
      "description":"Google's Material Design in XAML & WPF, for C# & VB.Net.",
      "language":"C#",
      "allStars":"2,285",
      "todaysStars":"8 stars today"
   },
   {
      "index":13,
      "name":"Hangfire",
      "owner":"HangfireIO",
      "description":"An easy way to perform background processing in your .NET and .NET Core applications",
      "language":"C#",
      "allStars":"2,275",
      "todaysStars":"8 stars today"
   },
   {
      "index":14,
      "name":"ToolGood.Words",
      "owner":"toolgood",
      "description":"一款高性能非法词(敏感词)检测组件，附带繁体简体互换，支持全角半角互换，获取拼音首字母，获取拼音字母等功能。",
      "language":"C#",
      "allStars":"36",
      "todaysStars":"7 stars today"
   },
   {
      "index":15,
      "name":"Mvc",
      "owner":"aspnet",
      "description":"ASP.NET Core MVC is a model view controller framework for building dynamic web sites with clean separation of concerns, including the merged MVC, Web API, and Web Pages w/ Razor.",
      "language":"C#",
      "allStars":"3,853",
      "todaysStars":"7 stars today"
   },
   {
      "index":16,
      "name":"dnSpy",
      "owner":"0xd4d",
      "description":".NET assembly editor, decompiler, and debugger",
      "language":"C#",
      "allStars":"1,430",
      "todaysStars":"7 stars today"
   },
   {
      "index":17,
      "name":"LiteDB",
      "owner":"mbdavid",
      "description":"LiteDB - A .NET NoSQL Document Store in a single data file -",
      "language":"C#",
      "allStars":"1,197",
      "todaysStars":"7 stars today"
   },
   {
      "index":18,
      "name":"SignalR",
      "owner":"aspnet",
      "description":"Incredibly simple real-time web for ASP.NET Core",
      "language":"C#",
      "allStars":"283",
      "todaysStars":"7 stars today"
   },
   {
      "index":19,
      "name":"coreclr",
      "owner":"dotnet",
      "description":"This repo contains the .NET Core runtime, called CoreCLR, and the base library, called mscorlib. It includes the garbage collector, JIT compiler, base .NET data types and many low-level classes.",
      "language":"C#",
      "allStars":"7,263",
      "todaysStars":"5 stars today"
   },
   {
      "index":20,
      "name":"VRTK",
      "owner":"thestonefox",
      "description":"A productive VR Toolkit for rapidly building VR solutions in Unity3d.",
      "language":"C#",
      "allStars":"1,156",
      "todaysStars":"5 stars today"
   },
   {
      "index":21,
      "name":"WeiXinMPSDK",
      "owner":"JeffreySu",
      "description":"微信公众平台SDK Senparc.Weixin for C#，支持.NET Framework及.NET Core。已支持微信公众号、小程序、企业号、企业微信、开放平台、微信支付、JSSDK、微信周边。 WeChat SDK for C#.",
      "language":"C#",
      "allStars":"1,638",
      "todaysStars":"5 stars today"
   },
   {
      "index":22,
      "name":"mono",
      "owner":"mono",
      "description":"Mono open source ECMA CLI, C# and .NET implementation.",
      "language":"C#",
      "allStars":"5,193",
      "todaysStars":"6 stars today"
   },
   {
      "index":23,
      "name":"Wox",
      "owner":"Wox-launcher",
      "description":"Launcher for Windows, an alternative to Alfred and Launchy.",
      "language":"C#",
      "allStars":"4,611",
      "todaysStars":"6 stars today"
   },
   {
      "index":24,
      "name":"msbuild",
      "owner":"Microsoft",
      "description":"The Microsoft Build Engine (MSBuild) is the build platform for .NET and Visual Studio.",
      "language":"C#",
      "allStars":"2,852",
      "todaysStars":"5 stars today"
   }
]
```

**Comment by rhysd** [comment]
Looks nice :+1:

Could you make a PR for this?

**Comment by bumbeishvili** [comment]
Ok

**Comment by rhysd** [comment]
I published 0.1.0 including this feature.
